remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -132,6 +135,34 @@ public void addItem(RemixerItem remixerItem) {
     return list;
   }
 
+  /**
+   * Notifies all other instances of variables with the same key that the value has changed and
+   * informs the synchronization mechanism as well.
+   */
+  void onValueChanged(Variable variable) {
+    synchronizationMechanism.onValueChanged(variable);
+    List<RemixerItem> itemList = getItemsWithKey(variable.getKey());
+    for (RemixerItem item : itemList) {
+      if (item != variable) {
+        ((Variable) item).setValueWithoutNotifyingOthers(variable.getSelectedValue());
```

Comment:
I think separate implementations are still necessary, and I would avoid duplicating code by making this localvaluesyncing class the base class for all of them. 

The reason for this is that local-only and cloud mode kinda have different rules RE: source of truth and stuff like that, that would make the code really hard to read in a single configurable object.